### PR TITLE
[CARBONDATA-4126] Concurrent compaction failed with load on table.

### PR DESCRIPTION
**Pull request by Karan980**
### Why is this PR needed?
Concurrent compaction was failing when run in parallel with load. During load we acquire SegmentLock for a particular segment, and when this same lock we try to acquire during compaction, we were not able to acquire this lock and compaction fails.
  
 ### What changes were proposed in this PR?
Skipped compaction for segments for which we are not able to acquire the SegmentLock instead of throwing the exception.

    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12444/job/ApacheCarbonPRBuilder2.3/5453/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12444/job/ApacheCarbon_PR_Builder_2.4.5/3692/

**Comment by ydvpankaj99** [comment]
retest this please

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/3322/

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/5080/

**Comment by CarbonDataQA2** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/3309/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/5067/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12444/job/ApacheCarbonPRBuilder2.3/5454/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12444/job/ApacheCarbon_PR_Builder_2.4.5/3693/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonTableCompactor.scala [hunk 1]
```diff
@@ -92,19 +92,22 @@ class CarbonTableCompactor(
...
+          segmentLocks += segmentLock
+        } else {
+          LOGGER.warn(s"Failed to acquire lock on segment ${segmentId.getLoadName}, " +
+                      s"during compaction of table ${compactionModel.carbonTable.getQualifiedName}")
         }
-        segmentLocks += segmentLock
       }
       try {
-        scanSegmentsAndSubmitJob(loadsToMerge, compactedSegments, compactedLoad)
+        scanSegmentsAndSubmitJob(validSegments, compactedSegments, compactedLoad)
```
Comment (kunal642) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonTableCompactor.scala:110]: log the valid segments identified for compaction

Comment (kunal642) [reply integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonTableCompactor.scala:110]: please add a test case for this scenario

Comment (Karan980) [reply integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonTableCompactor.scala:110]: Log statement is already present inside scanSegmentAndSubmitJob api.

Comment (kunal642) [reply integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonTableCompactor.scala:110]: ok

**Comment by kunal642** [comment]
retest this please

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12444/job/ApacheCarbon_PR_Builder_2.4.5/3720/

**Comment by CarbonDataQA2** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12444/job/ApacheCarbonPRBuilder2.3/5484/

**Comment by kunal642** [comment]
LGTM

**Comment by kunal642** [comment]
LGTM

**Pull request closed without merging**